hashtypechecker.cpp

```diff
@@ -0,0 +1,122 @@
+#include "hashtypechecker.h"
+
+#include <QtDebug>
+
+HashTypeChecker::HashTypeChecker()
+{
+    connect(&m_john,SIGNAL(finished(int, QProcess::ExitStatus)),this,SLOT(startParsing()));
+    connect (&m_john, SIGNAL(readyReadStandardOutput()), this, SLOT(processOutput())); // connect process signals with your code
+}
+
+HashTypeChecker::~HashTypeChecker()
+{
+    terminate();
+}
+void HashTypeChecker::start(QString& pathToJohn, QString& pathToPwdFile)
+{
+    // We make sure last process is terminated correctly before
+    // loading a new password file.
+    terminate();
+    m_johnOutput.clear();
+    m_johnOutput.append(pathToPwdFile + "\n");
+    m_john.start(pathToJohn + " --show=types " + pathToPwdFile);
+}
+void HashTypeChecker::terminate(bool shouldProcessWorkToDate)
+{
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.terminate();
+    m_john.waitForFinished(500);
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.kill();
+
+    // Process what have been done so far from JohnOutput ..
+    if(shouldProcessWorkToDate)
```

Comment:
Are you sure that you aren't calling startParsing() twice for the explicit termination case? once for finished() signal and again here?